corgi/core/models.py

```diff
@@ -513,7 +521,78 @@ def get_ofuri(self) -> str:
     @property
     def manifest(self) -> str:
         """Return an SPDX-style manifest in JSON format."""
-        return ProductManifestFile(self).render_content()
+        doc = Document(spdx_id="SPDXRef-DOCUMENT", version=Version(2, 2), name=self.name)
+        doc.namespace = f"https://access.redhat.com/security/data/manifest/spdx/{self.name}"
+        doc.data_license = License.from_identifier("CC0-1.0")
+        doc.creation_info.add_creator(
+            Organization("Red Hat Product Security", "secalert@redhat.com")
+        )
+        doc.creation_info.set_created_now()
+        for component in self.get_latest_components():
+            comp_spdx_id = self.add_manifest_package(component, doc)
+            # TODO fix the SPDX relationship types to use RelationType class once
+            #  https://github.com/spdx/tools-python/pull/207 is fixed
+            self.add_relationships(
+                comp_spdx_id,
+                component,
+                doc,
+                ComponentNode.ComponentNodeType.PROVIDES,
+                "CONTAINED_BY",
+            )
+            self.add_relationships(
+                comp_spdx_id,
+                component,
+                doc,
+                ComponentNode.ComponentNodeType.PROVIDES_DEV,
+                "DEV_DEPENDENCY_OF",
+            )
+
+        out = StringIO()
+        messages = ErrorMessages()
+        doc.validate(messages)
+        if messages:
+            logger.debug("Manifest document for stream %s is Invalid", self.name)
+            for message in messages:
+                logger.debug(message)
+        # TODO fix validation errors related to pg_verif_code being None (see test_manifests.py)
+        write_document(doc, out, validate=False)
+        return out.getvalue()
+
+    def add_relationships(self, comp_spdx_id, component, doc, component_type, relationship_type):
+        for descendant in component.cnodes.first().get_descendants().filter(type=component_type):
```

Comment:
This looks like a bug - if the component is in multiple trees, we only output descendants of the first one. Do we need descendants for a tree that relates to a particular product (the one we're manifesting), or descendants in all trees instead?

Reply:
I guess it's more robust to use provides array of component and lookup the components using their purls again.